src/Microsoft.ML.TimeSeries/SrCnnAnomalyDetectionBase.cs

```diff
@@ -292,6 +358,282 @@ private Single CalculateSocre(Single mag, Single avgMag)
                     }
                     return (float)(Math.Abs(mag - avgMag) / safeDivisor);
                 }
+
+                private List<int> GetAnomalyIndex(List<Single> magList, List<Single> avgMagList)
+                {
+                    List<int> anomalyIdxList = new List<int>();
+                    for (var i = 0; i < magList.Count; ++i)
+                    {
+                        double safeDivisor = avgMagList[i];
```

Comment:
>avgMagList[i] [](start = 45, length = 13)

Can this be negative? If yes, then you need to check the absolute value of `safeDivisor`.